flink-connector-aws-kinesis-firehose/archunit-violations/a6cbd99c-b115-447a-8f19-43c1094db549

```diff
@@ -2,5 +2,11 @@ org.apache.flink.connector.firehose.sink.KinesisFirehoseSinkITCase does not sati
 * reside in a package 'org.apache.flink.runtime.*' and contain any fields that are static, final, and of type InternalMiniClusterExtension and annotated with @RegisterExtension\
 * reside outside of package 'org.apache.flink.runtime.*' and contain any fields that are static, final, and of type MiniClusterExtension and annotated with @RegisterExtension\
 * reside in a package 'org.apache.flink.runtime.*' and is annotated with @ExtendWith with class InternalMiniClusterExtension\
+* reside outside of package 'org.apache.flink.runtime.*' and is annotated with @ExtendWith with class MiniClusterExtension\
+ or contain any fields that are public, static, and of type MiniClusterWithClientResource and final and annotated with @ClassRule or contain any fields that is of type MiniClusterWithClientResource and public and final and not static and annotated with @Rule
+org.apache.flink.connector.firehose.sink.KinesisFirehoseSinkITCase does not satisfy: only one of the following predicates match:\
+* reside in a package 'org.apache.flink.runtime.*' and contain any fields that are static, final, and of type InternalMiniClusterExtension and annotated with @RegisterExtension\
+* reside outside of package 'org.apache.flink.runtime.*' and contain any fields that are static, final, and of type MiniClusterExtension and annotated with @RegisterExtension or are , and of type MiniClusterTestEnvironment and annotated with @TestEnv\
+* reside in a package 'org.apache.flink.runtime.*' and is annotated with @ExtendWith with class InternalMiniClusterExtension\
```

Comment:
would be great if you link to the corresponding commit/jira which is the reason of that?

Reply:
Good suggestion, FLINK-31804 is the main reason. Perhaps we should indeed backport it to `release-1.17` and `release-1.16`(not checking if it is necessary). But before we reach an agreement on it, the workaround solution can first allow the connector to run the nightly test normally.

Reply:
But as the comment in issue: Adding new text of violation in ArchUnit store does not seem like a good solution since ArchUnit updates ignored violations by deleting ones that are not encountered during build. 

We generally do not run `TestCodeArchitectureTest` on the IDE, so there may not be any issues at the moment. But in the long run, we have to decide whether we need to backport `FLINK-31804`.